libraries/provider_consul_client_source.rb

```diff
@@ -0,0 +1,32 @@
+class Chef::Provider::ConsulServiceSource < Chef::Provider::ConsulService
+  action :create do
+    include_recipe 'golang::default'
+
+    directory File.join(node['go']['gopath'], 'src/github.com/hashicorp') do
```

Comment:
It's not a good idea to consume node attributes inside a provider... pass this in through the resource interface. You can do type checking (kind_of: String) there as well. I'd also use a method to calculate the string.